### PR TITLE
Fixed hipSetDeviceFlags and deleted duplicate test.

### PR DESCRIPTION
**Pull request by SarbojitAMD**

### CODE REVIEW

File: src/hip_device.cpp [hunk 1]
```diff
@@ -382,28 +382,15 @@ hipError_t hipSetDeviceFlags(unsigned int flags) {
     if (ctx) {
         auto* deviceHandle = ctx->getDevice();
         if (deviceHandle->_state == 0) {
```
Comment (emankov) [inline src/hip_device.cpp]: Make it, please, a single if statement with 2 null checks.

File: src/hip_device.cpp [hunk 2]
```diff
@@ -382,28 +382,15 @@ hipError_t hipSetDeviceFlags(unsigned int flags) {
     if (ctx) {
```
Comment (emankov) [inline src/hip_device.cpp]: Please, revise comments.

File: src/hip_device.cpp [hunk 3]
```diff
@@ -382,28 +382,15 @@ hipError_t hipSetDeviceFlags(unsigned int flags) {
...
             unsigned supportedFlags =
                 hipDeviceScheduleMask | hipDeviceMapHost | hipDeviceLmemResizeToMax;
 
             if (flags & (~supportedFlags)) {
                 e = hipErrorInvalidValue;
+            }else{
+                ctx->_ctxFlags = ctx->_ctxFlags | flags;
             }
+
         } else {
```
Comment (emankov) [inline src/hip_device.cpp]: Formatting.

**Comment by SarbojitAMD** [comment]
Have implemented in HIP-Clang hence closing it for now. If needed in HCC will resume it back.

**Pull request closed without merging**